control-plane/csi-controller/src/controller.rs

```diff
@@ -233,15 +241,20 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
             }
         };
 
+        // Check filesystem type.
+        if !valid_fs_type(args.parameters.get("fsType")) {
+            return Err(Status::invalid_argument("Invalid filesystem type"));
+        }
+
         // Check storage protocol.
-        let protocol = match args.parameters.get("protocol") {
-            Some(p) => p.to_string(),
-            None => return Err(Status::invalid_argument("Missing storage protocol")),
+        let protocol = match parse_protocol(args.parameters.get("protocol")) {
+            Ok(p) => p,
+            Err(e) => return Err(e),
```

Comment:
you can also use `?` here to avoid the match.

Reply:
Done.